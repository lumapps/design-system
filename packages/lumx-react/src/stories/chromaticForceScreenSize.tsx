/* eslint-disable */
import React from 'react';
import isChromatic from 'chromatic/isChromatic';

/** Story decorator used to force a minimum screen size when running in chromatic */
export const chromaticForceScreenSize = (story: any) =>
    isChromatic() ? <div style={{ minWidth: 1200, minHeight: 800 }}>{story()}</div> : story();
