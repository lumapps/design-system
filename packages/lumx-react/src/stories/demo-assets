../../../site-demo/static/demo-assets