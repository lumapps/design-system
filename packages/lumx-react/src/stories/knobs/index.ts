export * from './image';
export * from './lorem';
