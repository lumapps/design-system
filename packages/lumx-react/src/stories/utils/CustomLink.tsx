/* eslint-disable jsx-a11y/anchor-has-content,jsx-a11y/anchor-is-valid */
import React from 'react';

/**
 * Example custom link that can be used in linkAs props.
 */
export const CustomLink: any = ({ ...props }) => <a href="#" data-custom-link {...props} />;
