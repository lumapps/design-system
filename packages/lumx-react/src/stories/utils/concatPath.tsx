export type PathPart = string | number;
export type Path = PathPart | Array<PathPart>;

/**
 * Concat flatten object path
 *
 * @example concatPath('foo', 'bar') // => 'foo.bar'
 * @example concatPath(['foo', 0]) // => 'foo[0]'
 * @example concatPath('foo', 0, ['bar']) // => 'foo[0].bar'
 */
export const concatPath = (...prefix: Path[]) => {
    const [first, ...rest] = prefix.flat();
    return rest.reduce<string>((acc, part) => {
        if (typeof part === 'number') return `${acc}[${part}]`;
        return `${acc}.${part}`;
    }, String(first));
};
