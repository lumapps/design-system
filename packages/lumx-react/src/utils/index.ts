/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
