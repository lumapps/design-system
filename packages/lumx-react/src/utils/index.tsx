export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
