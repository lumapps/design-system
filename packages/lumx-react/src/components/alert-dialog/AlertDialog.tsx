import React, { forwardRef } from 'react';

import classNames from 'classnames';

import {
    DialogProps,
    Dialog,
    Button,
    Emphasis,
    ColorPalette,
    Icon,
    Size,
    Kind,
    Toolbar,
    ButtonProps,
} from '@lumx/react';

import { mdiAlert, mdiAlertCircle, mdiCheckCircle, mdiInformation } from '@lumx/icons/';
import { uid } from 'uid';
import { Comp, getRootClassName, handleBasicClasses } from '@lumx/react/utils';

export interface AlertDialogProps extends Omit<DialogProps, 'header' | 'footer'> {
    /** Message variant. */
    kind?: Kind;
    /** Dialog title. */
    title?: string;
    /** Props forwarded to the confirm button */
    confirmProps: ButtonProps & {
        onClick(): void;
        label: string;
    };
    /**
     * Props forwarded to the cancel button.
     * Will not render a cancel button if undefined.
     */
    cancelProps?: ButtonProps & {
        onClick(): void;
        label: string;
    };
}

/**
 * Associative map from message kind to color and icon.
 */
const CONFIG = {
    [Kind.error]: { color: ColorPalette.red, icon: mdiAlert },
    [Kind.info]: { color: ColorPalette.blue, icon: mdiInformation },
    [Kind.success]: { color: ColorPalette.green, icon: mdiCheckCircle },
    [Kind.warning]: { color: ColorPalette.yellow, icon: mdiAlertCircle },
};

/**
 * Component display name.
 */
const COMPONENT_NAME = 'AlertDialog';

/**
 * Component default class name and class prefix.
 */
const CLASSNAME = getRootClassName(COMPONENT_NAME);

/**
 * Component default props.
 */
const DEFAULT_PROPS: Partial<DialogProps> = {
    size: Size.tiny,
    kind: Kind.info,
};

/**
 * AlertDialog component.
 *
 * An alert dialog is a modal dialog that interrupts the user's workflow to
 * communicate an important message and acquire a response.
 *
 * It should not have a complex content.
 * Children of this component should only be strings, paragraphs or links.
 */
export const AlertDialog: Comp<AlertDialogProps, HTMLDivElement> = forwardRef((props, ref) => {
    const {
        id,
        title,
        className,
        cancelProps,
        confirmProps,
        kind,
        size,
        dialogProps,
        children,
        ...forwardedProps
    } = props;

    const cancelButtonRef = React.useRef(null);
    const confirmationButtonRef = React.useRef(null);
    const { color, icon } = CONFIG[kind as Kind] || {};

    // Define a unique ID to target title and description for aria attributes.
    const uniqueId = React.useMemo(() => id || uid(), [id]);
    const titleId = `${uniqueId}-title`;
    const descriptionId = `${uniqueId}-description`;

    // If content is a string, set in a paragraph.
    const DescriptionElement = typeof children === 'string' ? 'p' : 'div';

    const { label: confirmLabel, onClick: confirmOnClick, ...forwardedConfirmProps } = confirmProps;
    const { label: cancelLabel, onClick: cancelOnClick, ...forwardedCancelProps } = cancelProps || {};

    return (
        <Dialog
            ref={ref}
            focusElement={cancelProps ? cancelButtonRef : confirmationButtonRef}
            size={size}
            dialogProps={{
                id: uniqueId,
                role: 'alertdialog',
                'aria-labelledby': titleId,
                'aria-describedby': descriptionId,
                ...dialogProps,
            }}
            className={classNames(
                className,
                handleBasicClasses({
                    kind,
                    prefix: CLASSNAME,
                }),
            )}
            {...forwardedProps}
        >
            <header>
                <Toolbar
                    className="lumx-spacing-margin-horizontal"
                    before={<Icon icon={icon} size={Size.s} color={color} />}
                    label={
                        <h2 id={titleId} className="lumx-typography-title">
                            {title}
                        </h2>
                    }
                />
            </header>

            {children && (
                <DescriptionElement
                    id={descriptionId}
                    className="lumx-typography-body2 lumx-spacing-padding-vertical-big lumx-spacing-padding-horizontal-huge"
                >
                    {children}
                </DescriptionElement>
            )}

            <footer>
                <Toolbar
                    className="lumx-spacing-margin-horizontal"
                    after={
                        <>
                            {cancelProps && (
                                <Button
                                    {...forwardedCancelProps}
                                    ref={cancelButtonRef}
                                    emphasis={Emphasis.medium}
                                    onClick={cancelOnClick}
                                >
                                    {cancelLabel}
                                </Button>
                            )}
                            <Button
                                {...forwardedConfirmProps}
                                ref={confirmationButtonRef}
                                color={color}
                                className="lumx-spacing-margin-left-regular"
                                onClick={confirmOnClick}
                            >
                                {confirmLabel}
                            </Button>
                        </>
                    }
                />
            </footer>
        </Dialog>
    );
});

AlertDialog.displayName = COMPONENT_NAME;
AlertDialog.className = CLASSNAME;
AlertDialog.defaultProps = DEFAULT_PROPS;
