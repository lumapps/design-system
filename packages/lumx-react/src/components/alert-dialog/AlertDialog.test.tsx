import React, { ReactElement } from 'react';

import { mount, shallow } from 'enzyme';
import 'jest-enzyme';

import { commonTestsSuite, itShouldRenderStories, Wrapper } from '@lumx/react/testing/utils';
import { AlertDialog, AlertDialogProps } from './AlertDialog';
import * as stories from './AlertDialog.stories';

jest.mock('uid', () => ({ uid: () => 'uid' }));

const CLASSNAME = AlertDialog.className as string;

/**
 * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
 */
const setup = ({ ...propsOverride }: Partial<AlertDialogProps> = {}, shallowRendering = true) => {
    const props: AlertDialogProps = {
        title: 'Alert',
        description: 'Deserunt et sunt qui consequat sint sit.',
        confirmProps: { onClick: () => alert('confirm'), label: 'OK' },
        ...propsOverride,
    };
    const renderer: (el: ReactElement) => Wrapper = shallowRendering ? shallow : mount;
    const wrapper: Wrapper = renderer(<AlertDialog {...props} />);

    return {
        AlertDialog: wrapper.find(`Dialog`),
        props,
        wrapper,
    };
};

describe(`<${AlertDialog.displayName}>`, () => {
    // 1. Test render via snapshot.
    describe('Snapshots and structure', () => {
        itShouldRenderStories(stories, AlertDialog);
    });

    // Common tests suite.
    commonTestsSuite(setup, { className: 'AlertDialog', prop: 'AlertDialog' }, { className: CLASSNAME });
});
