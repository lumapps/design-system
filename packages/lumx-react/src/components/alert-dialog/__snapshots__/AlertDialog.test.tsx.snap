// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertDialog> Snapshots and structure should render story 'Default' 1`] = `
<Dialog
  className="lumx-alert-dialog lumx-alert-dialog--kind-info"
  dialogProps={
    Object {
      "aria-describedby": "uid-description",
      "aria-labelledby": "uid-title",
      "id": "uid",
      "role": "alertdialog",
    }
  }
  focusElement={
    Object {
      "current": null,
    }
  }
  isOpen={true}
  onClose={[Function]}
  parentElement={
    Object {
      "current": undefined,
    }
  }
  size="tiny"
>
  <header>
    <Toolbar
      before={
        <Icon
          color="blue"
          icon="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          size="s"
        />
      }
      className="lumx-spacing-margin-horizontal"
      label={
        <h2
          className="lumx-typography-title"
          id="uid-title"
        >
          Default (info)
        </h2>
      }
    />
  </header>
  <p
    className="lumx-typography-body2 lumx-spacing-padding-vertical-big lumx-spacing-padding-horizontal-huge"
    id="uid-description"
  >
    Consequat deserunt officia aute laborum tempor anim sint est.
  </p>
  <footer>
    <Toolbar
      after={
        <React.Fragment>
          <Button
            className="lumx-spacing-margin-left-regular"
            color="blue"
            emphasis="high"
            onClick={[Function]}
            size="m"
            theme="light"
          >
            Confirm
          </Button>
        </React.Fragment>
      }
      className="lumx-spacing-margin-horizontal"
    />
  </footer>
</Dialog>
`;

exports[`<AlertDialog> Snapshots and structure should render story 'Error' 1`] = `
<Dialog
  className="lumx-alert-dialog lumx-alert-dialog--kind-error"
  dialogProps={
    Object {
      "aria-describedby": "alert-dialog-description",
      "aria-labelledby": "alert-dialog-title",
      "id": "alert-dialog",
      "role": "alertdialog",
    }
  }
  focusElement={
    Object {
      "current": null,
    }
  }
  isOpen={true}
  onClose={[Function]}
  parentElement={
    Object {
      "current": undefined,
    }
  }
  size="tiny"
>
  <header>
    <Toolbar
      before={
        <Icon
          color="red"
          icon="M13 14H11V9H13M13 18H11V16H13M1 21H23L12 2L1 21Z"
          size="s"
        />
      }
      className="lumx-spacing-margin-horizontal"
      label={
        <h2
          className="lumx-typography-title"
          id="alert-dialog-title"
        >
          Error
        </h2>
      }
    />
  </header>
  <p
    className="lumx-typography-body2 lumx-spacing-padding-vertical-big lumx-spacing-padding-horizontal-huge"
    id="alert-dialog-description"
  >
    Consequat deserunt officia aute laborum tempor anim sint est.
  </p>
  <footer>
    <Toolbar
      after={
        <React.Fragment>
          <Button
            className="lumx-spacing-margin-left-regular"
            color="red"
            emphasis="high"
            onClick={[Function]}
            size="m"
            theme="light"
          >
            OK
          </Button>
        </React.Fragment>
      }
      className="lumx-spacing-margin-horizontal"
    />
  </footer>
</Dialog>
`;

exports[`<AlertDialog> Snapshots and structure should render story 'RichDescription' 1`] = `
<Dialog
  className="lumx-alert-dialog lumx-alert-dialog--kind-info"
  dialogProps={
    Object {
      "aria-describedby": "alert-dialog-description",
      "aria-labelledby": "alert-dialog-title",
      "id": "alert-dialog",
      "role": "alertdialog",
    }
  }
  focusElement={
    Object {
      "current": null,
    }
  }
  isOpen={true}
  onClose={[Function]}
  parentElement={
    Object {
      "current": undefined,
    }
  }
  size="tiny"
>
  <header>
    <Toolbar
      before={
        <Icon
          color="blue"
          icon="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          size="s"
        />
      }
      className="lumx-spacing-margin-horizontal"
      label={
        <h2
          className="lumx-typography-title"
          id="alert-dialog-title"
        >
          With Rich Description
        </h2>
      }
    />
  </header>
  <div
    className="lumx-typography-body2 lumx-spacing-padding-vertical-big lumx-spacing-padding-horizontal-huge"
    id="alert-dialog-description"
  >
    Amet ut elit dolore irure mollit 
    <strong>
      sunt culpa esse
    </strong>
    .
    <br />
    Ea ut Lorem.
    <br />
    <Link
      href="https://example.com"
      target="_blank"
    >
      Link
    </Link>
  </div>
  <footer>
    <Toolbar
      after={
        <React.Fragment>
          <Button
            className="lumx-spacing-margin-left-regular"
            color="blue"
            emphasis="high"
            onClick={[Function]}
            size="m"
            theme="light"
          >
            OK
          </Button>
        </React.Fragment>
      }
      className="lumx-spacing-margin-horizontal"
    />
  </footer>
</Dialog>
`;

exports[`<AlertDialog> Snapshots and structure should render story 'Success' 1`] = `
<Dialog
  className="lumx-alert-dialog lumx-alert-dialog--kind-success"
  dialogProps={
    Object {
      "aria-describedby": "alert-dialog-description",
      "aria-labelledby": "alert-dialog-title",
      "id": "alert-dialog",
      "role": "alertdialog",
    }
  }
  focusElement={
    Object {
      "current": null,
    }
  }
  isOpen={true}
  onClose={[Function]}
  parentElement={
    Object {
      "current": undefined,
    }
  }
  size="tiny"
>
  <header>
    <Toolbar
      before={
        <Icon
          color="green"
          icon="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z"
          size="s"
        />
      }
      className="lumx-spacing-margin-horizontal"
      label={
        <h2
          className="lumx-typography-title"
          id="alert-dialog-title"
        >
          Success
        </h2>
      }
    />
  </header>
  <p
    className="lumx-typography-body2 lumx-spacing-padding-vertical-big lumx-spacing-padding-horizontal-huge"
    id="alert-dialog-description"
  >
    Consequat deserunt officia aute laborum tempor anim sint est.
  </p>
  <footer>
    <Toolbar
      after={
        <React.Fragment>
          <Button
            className="lumx-spacing-margin-left-regular"
            color="green"
            emphasis="high"
            onClick={[Function]}
            size="m"
            theme="light"
          >
            OK
          </Button>
        </React.Fragment>
      }
      className="lumx-spacing-margin-horizontal"
    />
  </footer>
</Dialog>
`;

exports[`<AlertDialog> Snapshots and structure should render story 'Warning' 1`] = `
<Dialog
  className="lumx-alert-dialog lumx-alert-dialog--kind-warning"
  dialogProps={
    Object {
      "aria-describedby": "alert-dialog-description",
      "aria-labelledby": "alert-dialog-title",
      "id": "alert-dialog",
      "role": "alertdialog",
    }
  }
  focusElement={
    Object {
      "current": null,
    }
  }
  isOpen={true}
  onClose={[Function]}
  parentElement={
    Object {
      "current": undefined,
    }
  }
  size="tiny"
>
  <header>
    <Toolbar
      before={
        <Icon
          color="yellow"
          icon="M13,13H11V7H13M13,17H11V15H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          size="s"
        />
      }
      className="lumx-spacing-margin-horizontal"
      label={
        <h2
          className="lumx-typography-title"
          id="alert-dialog-title"
        >
          Warning
        </h2>
      }
    />
  </header>
  <p
    className="lumx-typography-body2 lumx-spacing-padding-vertical-big lumx-spacing-padding-horizontal-huge"
    id="alert-dialog-description"
  >
    Consequat deserunt officia aute laborum tempor anim sint est.
  </p>
  <footer>
    <Toolbar
      after={
        <React.Fragment>
          <Button
            className="lumx-spacing-margin-left-regular"
            color="yellow"
            emphasis="high"
            onClick={[Function]}
            size="m"
            theme="light"
          >
            OK
          </Button>
        </React.Fragment>
      }
      className="lumx-spacing-margin-horizontal"
    />
  </footer>
</Dialog>
`;

exports[`<AlertDialog> Snapshots and structure should render story 'WithCancel' 1`] = `
<Dialog
  className="lumx-alert-dialog lumx-alert-dialog--kind-info"
  dialogProps={
    Object {
      "aria-describedby": "alert-dialog-description",
      "aria-labelledby": "alert-dialog-title",
      "id": "alert-dialog",
      "role": "alertdialog",
    }
  }
  focusElement={
    Object {
      "current": null,
    }
  }
  isOpen={true}
  onClose={[Function]}
  parentElement={
    Object {
      "current": undefined,
    }
  }
  size="tiny"
>
  <header>
    <Toolbar
      before={
        <Icon
          color="blue"
          icon="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          size="s"
        />
      }
      className="lumx-spacing-margin-horizontal"
      label={
        <h2
          className="lumx-typography-title"
          id="alert-dialog-title"
        >
          With Cancel
        </h2>
      }
    />
  </header>
  <p
    className="lumx-typography-body2 lumx-spacing-padding-vertical-big lumx-spacing-padding-horizontal-huge"
    id="alert-dialog-description"
  >
    Consequat deserunt officia aute laborum tempor anim sint est.
  </p>
  <footer>
    <Toolbar
      after={
        <React.Fragment>
          <Button
            emphasis="medium"
            onClick={[Function]}
            size="m"
            theme="light"
          >
            Cancel
          </Button>
          <Button
            className="lumx-spacing-margin-left-regular"
            color="blue"
            emphasis="high"
            onClick={[Function]}
            size="m"
            theme="light"
          >
            OK
          </Button>
        </React.Fragment>
      }
      className="lumx-spacing-margin-horizontal"
    />
  </footer>
</Dialog>
`;

exports[`<AlertDialog> Snapshots and structure should render story 'WithForwardedProps' 1`] = `
<Dialog
  className="lumx-alert-dialog lumx-alert-dialog--kind-info"
  dialogProps={
    Object {
      "aria-describedby": "alert-dialog-description",
      "aria-labelledby": "alert-dialog-title",
      "id": "alert-dialog",
      "role": "alertdialog",
    }
  }
  focusElement={
    Object {
      "current": null,
    }
  }
  isOpen={true}
  onClose={[Function]}
  parentElement={
    Object {
      "current": undefined,
    }
  }
  size="tiny"
>
  <header>
    <Toolbar
      before={
        <Icon
          color="blue"
          icon="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          size="s"
        />
      }
      className="lumx-spacing-margin-horizontal"
      label={
        <h2
          className="lumx-typography-title"
          id="alert-dialog-title"
        >
          With Forwarded Props
        </h2>
      }
    />
  </header>
  <p
    className="lumx-typography-body2 lumx-spacing-padding-vertical-big lumx-spacing-padding-horizontal-huge"
    id="alert-dialog-description"
  >
    Consequat deserunt officia aute laborum tempor anim sint est.
  </p>
  <footer>
    <Toolbar
      after={
        <React.Fragment>
          <Button
            emphasis="medium"
            onClick={[Function]}
            size="m"
            style={
              Object {
                "color": "blue",
              }
            }
            theme="light"
          >
            Cancel
          </Button>
          <Button
            className="lumx-spacing-margin-left-regular"
            color="blue"
            emphasis="high"
            onClick={[Function]}
            size="m"
            style={
              Object {
                "color": "red",
              }
            }
            theme="light"
          >
            OK
          </Button>
        </React.Fragment>
      }
      className="lumx-spacing-margin-horizontal"
    />
  </footer>
</Dialog>
`;
